presto-main/src/main/java/com/facebook/presto/connector/system/CatalogSystemTable.java

```diff
@@ -71,7 +75,12 @@ public RecordCursor cursor(ConnectorTransactionHandle transactionHandle, Connect
         TransactionId transactionId = checkType(transactionHandle, GlobalSystemTransactionHandle.class, "transactionHandle").getTransactionId();
         Builder table = InMemoryRecordSet.builder(CATALOG_TABLE);
         for (Map.Entry<String, ConnectorId> entry : transactionManager.getCatalogNames(transactionId).entrySet()) {
-            table.addRow(entry.getKey(), entry.getValue().toString());
+            try {
+                accessControl.checkCanAccessCatalog(transactionId, session.getIdentity(), entry.getKey());
+                table.addRow(entry.getKey(), entry.getValue().toString());
+            }
+            catch (AccessDeniedException ignore) {
```

Comment:
why don't we let the `AccessDeniedException` propagate instead of swallowing it?